src/feedClass.ts

```diff
@@ -0,0 +1,281 @@
+import * as _ from 'lodash'
+import Gun from 'gun'
+import InterfaceFeed from './interfaces/interfaceFeed'
+import {Feed, FeedInfo, FeedItem, Feeds} from './types'
+import gunDB from './lib/gundb'
+import logger from './lib/logger'
+
+
+export default class feedClass implements InterfaceFeed {
+    gun = gunDB;
+    gunUser = this.gun.user();
+    feeds = null;
+
+    feedItemDataFields = [
+        'rom',
+        'amount',
+        'code',
+        'reason',
+        'counterPartyDisplayName',
+        'paymentLink',
+        'otplStatus',
+        'customName',
+        'subtitle',
+    ];
+    feedInfoFields = [
+        'name',
+        'id',
+        'owner',
+    ];
+    feedItemFields = [
+        'id',
+        'type',
+        'status',
+        'createdDate',
+        'data',
+    ];
+
+    /**
+     * Auth in gun and return all feeds
+     *
+     * @param {string} username
+     * @param {string} password
+     *
+     * @return Feeds | boolean
+     */
+    async login(username: string, password: string): Promise<Feeds | boolean> {
+        logger.info('class login');
+        const user = await this.gunAuthInitFeed(username, password);
+        return (user)
+    }
+
+    /**
+     * add item to feed
+     * @param {string} feedInfoName
+     * @param {FeedItem} feedItem
+     */
+    async addItem(feedInfoName: string, feedItem: FeedItem): Promise<boolean> {
+        const date = new Date();
+        const day = `${date.toISOString().slice(0, 10)}`;
+
+        await this.feeds.get(feedInfoName).get('items').get(day).put(feedItem);
+        return true
+    }
+
+    /**
+     * Add private item to feed
+     * @param {string} feedInfoName
+     * @param {FeedItem} feedItem
+     * @param {string} privateKey
+     */
+    async addPrivateItem(feedInfoName: string, feedItem: FeedItem, privateKey: string): Promise<boolean> {
+        const date = new Date();
+        const day = `${date.toISOString().slice(0, 10)}`;
+
+        await this.feeds.get(feedInfoName).get('privateItems').get(privateKey).get(day).put(feedItem);
+        return true
+    }
+
+    /**
+     * Create new feed
+     *
+     * @param {Feed} feed
+     */
+    async createFeed(feed: Feed): Promise<Feed | any> {
+        logger.info('class createFeed');
+
+        feed.info.owner = await this.getOwnerSoul();
+        const gunNewFeed = this.feeds.get(feed.info.name).put(feed);
+        const id = await this.getSoul(gunNewFeed);
+        await gunNewFeed.get('info').get('id').put(id);
```

Comment:
there is a .path() shortener allows to avoid log .get() chains
https://gun.eco/docs/API#-a-name-path-a-gun-path-key-